File: 804-rotated-digits/rotated-digits.py
class Solution:
    def rotatedDigits(self, n: int) -> int:
        ans = 0
        mirr_rotate = ['2','5','6','9']

        for num in range(1,n+1):
            mirror_digit_found = False
            invalid_digit_found = False
            for digit in str(num):
                if digit in mirr_rotate:
                    mirror_digit_found = True
                elif digit in ['3','4','7']:
                    invalid_digit_found = True
                    break
            if mirror_digit_found and invalid_digit_found == False:
                ans+=1

        return ans

File: 804-rotated-digits/rotated-digits.java
class Solution {
    public int rotatedDigits(int n) {
        int ans = 0;
        
        for (int i = 1; i <= n; i++) {
            String num = String.valueOf(i);
            boolean invalidDigitFound = false;
            boolean rotatingDigitFound = false;

            for (char digit : num.toCharArray()) {
                if (digit == '2' || digit == '5' || digit == '6' || digit == '9') {
                    rotatingDigitFound = true;
                } else if (digit == '3' || digit == '4' || digit == '7'){
                    invalidDigitFound = true;
                    break;
                }
            }
            if (rotatingDigitFound && !invalidDigitFound){
                ans+=1;
            }
        }
        return ans;
    }
}